src/components/OptionListContextProvider.tsx

```diff
@@ -0,0 +1,142 @@
+import React, {createContext, useCallback, useContext, useEffect, useMemo, useRef, useState} from 'react';
+import {withOnyx} from 'react-native-onyx';
+import type {OnyxCollection} from 'react-native-onyx';
+import * as OptionsListUtils from '@libs/OptionsListUtils';
+import type {OptionList} from '@libs/OptionsListUtils';
+import * as ReportUtils from '@libs/ReportUtils';
+import ONYXKEYS from '@src/ONYXKEYS';
+import type {Report} from '@src/types/onyx';
+import {usePersonalDetails} from './OnyxProvider';
+
+type OptionsListContextProps = {
+    /** List of options for reports and personal details */
+    options: OptionList;
+    /** Function to initialize the options */
+    initializeOptions: () => void;
+    /** Flag to check if the options are initialized */
+    areOptionsInitialized: boolean;
+};
+
+type OptionsListProviderOnyxProps = {
+    /** Collection of reports */
+    reports: OnyxCollection<Report>;
+};
+
+type OptionsListProviderProps = OptionsListProviderOnyxProps & {
+    /** Actual content wrapped by this component */
+    children: React.ReactNode;
+};
+
+const OptionsListContext = createContext<OptionsListContextProps>({
+    options: {
+        reports: [],
+        personalDetails: [],
+    },
+    initializeOptions: () => {},
+    areOptionsInitialized: false,
+});
+
+function OptionsListContextProvider({reports, children}: OptionsListProviderProps) {
+    const areOptionsInitialized = useRef(false);
+    const [options, setOptions] = useState<OptionList>({
+        reports: [],
+        personalDetails: [],
+    });
+    const personalDetails = usePersonalDetails();
+
+    useEffect(() => {
```

Comment:
Coming from https://github.com/Expensify/App/issues/40493, we missed updating the cache when a report is deleted.